libsolidity/formal/ModelChecker.cpp

```diff
@@ -43,7 +44,32 @@ ModelChecker::ModelChecker(
 
 void ModelChecker::analyze(SourceUnit const& _source)
 {
-	if (!_source.annotation().experimentalFeatures.count(ExperimentalFeature::SMTChecker))
+	// TODO This should be removed for 0.9.0.
+	if (_source.annotation().experimentalFeatures.count(ExperimentalFeature::SMTChecker))
+	{
+		PragmaDirective const* smtPragma = nullptr;
+		for (auto node: _source.nodes())
+			if (auto pragma = dynamic_pointer_cast<PragmaDirective>(node))
+				if (
+					pragma->literals().size() >= 2 &&
+					pragma->literals().at(1) == "SMTChecker"
+				)
+				{
+					smtPragma = pragma.get();
+					break;
+				}
+		solAssert(smtPragma, "");
+		m_errorReporter.warning(
+			5523_error,
+			smtPragma->location(),
+			"The SMTChecker pragma has been deprecated and will be removed in the future. "
+			"Please use the \"model checker engine\" compiler setting to activate the SMTChecker instead. "
+			"If the pragma is enabled, all engines will be used."
+		);
+		m_settings.engine = ModelCheckerEngine::All();
```

Comment:
It looks like it could randomly enable SMT checking for some files that do not have the pragma. If `ModelChecker` instance is reused rather than recreated for each source unit, one file containing this pragma will enable it for all subsequent files. And the result will depend on the order in which files are processed.

Reply:
That was the decision in the call, to run for all if one file has the pragma. It's also fine with me to rediscuss it.

Reply:
I think that running for all files would be fine but but this does not necessarily work like that. If you have 10 files and the third one has the pragma, it will not run for the first two files. But might then suddenly run if user reorders his imports.

Reply:
Hmm that's true

Reply:
In fact the pragma behavior in this PR is to report anything from any file that has the pragma *or* is directly or indirectly imported by a file that has the pragma

Reply:
Ah no, it's this /\ + what you said first